src/App.jsx

```diff
@@ -14,45 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'alp';
+const SORT_BY_LENGTH = 'lng';
+const REVERSE = true;
+
+function sortGoods(goods, { sortCondition, reverseGoods }) {
+  const sortedGoods = [...goods];
+
+  if (sortCondition) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortCondition) {
+        case SORT_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseGoods) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [sortCondition, setSortCondition] = useState('');
+  const [reverseGoods, setReverseGoods] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, {
+    sortCondition,
+    reverseGoods,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetButtonIsVisible = sortCondition || reverseGoods;
+
+  function handleReverse(reverse) {
+    setReverseGoods(prev => (prev === reverse ? false : reverse));
+  }
```

Comment:
This function is overcomplicated. You can simplified it:
```suggestion
  function handleReverse() {
    setReverseGoods(prev => !prev);
  }
```